docs/agents/agent-development.md

```diff
@@ -237,14 +237,37 @@ Global labels can be specified via the environment variable `ELASTIC_APM_GLOBAL_
 
 ### Transactions
 
+#### Transaction outcome
+
+The optional `outcome` property denotes whether the transaction represents a success or a failure.
+The APM Server converts this to the [`event.outcome`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html) field.
+
+- `"failure"`: Indicates that this transaction describes a failed result.
+- `"success"`: Indicates that this transaction describes a successful result.
+- `"unknown"`:
+  TODO: does unknown make sense in the context of spans and transactions?
+  Would `unknown` apply to an HTTP 202 Accepted? Or a client span that puts a message into a queue?
+  In these cases we only know whether a message could successfully be enqueued but not if it could successfully be processed.
+  Or is this reserved to events that only represent the request part and where the response is a separate event?
+  A transaction contains information about both the request and the response.
+  If `unknown` doesn't apply to transactions it might make more sense to use a boolean as the datatype of this field in the intake API and let APM Server convert to `failure`/`success`. 
+- `null`/`undefined`: There's no information about the outcome of the transaction. \
+  Note that this is not equal to [`event.outcome: unknown`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-outcome.html#ecs-event-outcome-unknown)
+  as it would imply that the event is not a `"failure"`.
+  Not setting the `outcome`, or setting it to `null`, makes no statement on the outcome of the transaction.
+  This may be the case when tracking a custom transaction without explicitly setting an outcome.
```

Comment:
I am not 100% sure if having an outcome of `null` in this case helps us in some way. Any concerns of having this outcome as `unknown` as opposed to `null`? 

Asking because, In the context of page load for RUM the agent cannot determine the outcome of the transaction as  it's not available through any browser API's. Setting` unknown` feels better than having it as `null` for this type.

Reply:
That makes sense, We can default to "unknown" for page-load transactions and for some route-change transactions where we cannot determine the status codes. But not sure exposing an API for it. Are we planning to do any analysis based on the `outcome`?

Reply:
> Are we planning to do any analysis based on the outcome?

It will mostly be used to reason about error rates of a service or a transaction group.

Reply:
I've included the RUM use case in the `HTTP Transactions` section

Reply:
Nice, thanks 👍